### ISSUE TITLE
Documentation example `--extend-exclude ^/foo.py` does not exclude root file `foo.py`

### ISSUE THREAD

**Issue by ketozhang** [psf/black#2750]
**Describe the bug**
Black does not exclude the file`foo.py` root when asked to exclude it with `^/foo.py`. This particular example was used in the [documentation](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html?highlight=.venv#configuration-format):

> ```
> extend-exclude = '''
> # A regex preceded with ^/ will apply only to files and directories
> # in the root of the project.
> ^/foo.py  # exclude a file named foo.py in the root of the project (in addition to the defaults)
> '''
> ```

**To Reproduce**
Start with an empty directory

```sh
$ echo "s='test'" > foo.py

$ black --extend-exclude "^/foo.py" --check .
would reformat foo.py
Oh no! 💥 💔 💥
1 file would be reformatted.
```

Same behavior using `--exclude` flag

**Environment**

- Black's version: 21.12b0
- OS and Python version: OSX 11.6, Python 3.9.6

**Comment by ketozhang** [comment]
Actually this is more likely a documentation typo as the proper regex should be `^/foo\.py`

**Comment by ketozhang** [comment]
However, this does work if placed in `pyproject.toml`. Weird

The proper regex should escape the period `^/foo\.py`. Why is it that you need proper regex in commandline but not in the toml file?

**Comment by MarcoGorelli** [comment]
> However, this does work if placed in pyproject.toml. Weird

Yes, because then `black` is able to correctly determine where the root of the project it. It would also work to do `git init`

**Comment by ketozhang** [comment]
Oh that's right. I'd mistakenly assume black would use CWD as root when .git/ or pyproject.toml is missing.

**Comment by ketozhang** [comment]
To be fair, there is is no indication of this in `black --help`. I was surprised `^/` even worked in the first place (matching the style of `.gitignore`).

**Comment by MarcoGorelli** [comment]
> To be fair, there is is no indication of this in `black --help`. I was surprised `^/` even worked in the first place.

Just a suggestion - do you want to add a note to the docs? e.g.
```
# A regex preceded with ^/ will apply only to files and directories
# in the root of the project.
^/foo.py  # exclude a file named foo.py in the root of the project (in addition to the defaults)
# see "Where Black looks for the file" for how black determines where the root is
'''
```

**Comment by ketozhang** [comment]
I don't think that particular modification would help since for the case of pyproject file, the root is always the path of the pyproject file.

A section on general examples of exclusion patterns will help:

* `foo` exclude any file or directory if its path has this substring
* `^/foo$`, exclude file or directory on project root (see [Where Black looks for the file](#))
    Make sure to end with `$`; without it, directories like `foobar/` will be excluded. 
* `foo\/` exclude any directory if its path has this substring

Largely needed since everyone makes the mistake of attempting to use glob patterns (e.g., `\.\/foo` and `^foo\/` doesn't work as intended)

**Comment by JosEnerv** [comment]
extend-exclude does not do what I want it to do somehow : 
extend-exluded : `src\/models`

Verbose output : 

`src/models/something.py` ignored: matches the --extend-exclude regular expression
would reformat `src/models/something.py`

I don't get it.

**Comment by tomschr** [comment]
I had the same problem and circumvent it by using Git's `.gitignore` and adding my excludes:

```
# .gitignore file
/foo.py
```

If you want to ignore all Python files in root (but not `setup.py`, for example) you could add these lines:

```
/*.py
!/setup.py
```

I've prepared a small environment to show how it looks like (see below for details). When I run `black` in verbose mode I get:

```
$ black --verbose --check .
Identified `/tmp/gitblack` as project root containing a .git directory.
Sources to be formatted: "."
.git ignored: matches the --exclude regular expression
foo.py ignored: matches the .gitignore file content
setup.py already well formatted, good job.
src/foo/foo.py already well formatted, good job.

All done! ✨ 🍰 ✨
2 files would be left unchanged.

```

<details>
   <summary>Details about the test Git repo</summary>

```
$ mkdir -p /tmp/gitblack && cd /tmp/gitblack
$ git init .
$ mkdir -p src/foo/
$ touch foo.py src/foo/foo.py setup.py
$ cat << EOF > .gitignore
/*.py
!/setup.py
EOF
$ git add .gitignore setup.py src/foo/foo.py
$ git commit -m "Initial import"
```
</details>

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll submit a PR shortly.
